### PR TITLE
Add vsock seqpacket test

### PR DESCRIPTION
**Pull request by qcheng-redhat**
ID: 2052375

Signed-off-by: qcheng <qcheng@redhat.com>

### CODE REVIEW

**Comment by qcheng-redhat** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.upstream_testsuite.q35: PASS (1669.10 s)

**Comment by qcheng-redhat** [comment]
Hi @zhenyzha @PaulYuuu, please help review this patch. Thanks.

File: qemu/tests/vsock_seqpacket_test.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    guest_cid = vm.devices.get(vsock_dev).get_param("guest-cid")
+    host_cid = params["host_cid"]
+    port = random.randrange(1, 6000)
+    test_bin = params["testsuite_dir"] + "vsock_test"
+
+    try:
+        download_compile_upstream_testsuite(test, params, session)
+
+        # Scenario I: host = client, guest = server
+        LOG_JOB.info("Host as client, guest as server...")
```
Comment (PaulYuuu) [inline qemu/tests/vsock_seqpacket_test.py:69]: Should use `test.log` in the `run` function.

File: qemu/tests/cfg/vsock.cfg [hunk 2]
```diff
@@ -43,3 +43,14 @@
             ppc64, ppc64le:
                 virtio_dev_disable_legacy = on
                 virtio_dev_disable_modern = off
+        - seqpacket_test:
+             no RHEL.6 RHEL.7 RHEL.8
+             type = vsock_seqpacket_test
+             host_cid = 2
+             disable_firewall = "service iptables stop; systemctl stop firewalld.service"
+             rm_cmd = "cd /home/ && rm -rf linux"
+             clone_cmd = "${rm_cmd} && git clone https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git"
```
Comment (PaulYuuu) [inline qemu/tests/cfg/vsock.cfg:52]: Add `--depth=1` to reduce time.

File: qemu/tests/vsock_seqpacket_test.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
+import logging
+import random
+import time
+import aexpect
```
Comment (PaulYuuu) [inline qemu/tests/vsock_seqpacket_test.py:4]: `aexpect` is an avocado-framework module, not a standard module.

File: qemu/tests/vsock_seqpacket_test.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    host_ip = utils_net.get_host_ip_address(params)
+    guest_ip = vm.get_address()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    vsock_dev = params["vsocks"].split()[0]
```
Comment (PaulYuuu) [inline qemu/tests/vsock_seqpacket_test.py:59]:
```suggestion
    vsock_dev = params.objects("vsocks")[0]
```

File: qemu/tests/vsock_seqpacket_test.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        LOG_JOB.info("Host as server, guest as client...")
+        client_cmd = params["client_cmd"] % (test_bin, host_ip, port, host_cid)
+        server_cmd = params["server_cmd"] % (test_bin, port, guest_cid)
+        aexpect.Expect(
+            server_cmd,
+            auto_close=False,
+            output_func=utils_misc.log_line,
+            output_params=("vsock_%s_%s" % (guest_cid, port),))
+        time.sleep(5)
+        status, output = session.cmd_status_output(client_cmd)
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py:94]: When vsock_tests is executed in guest_cmd, no matter it fails or passes,  the server_cmd will be terminated.  Anyway I added a check here, please review.

File: qemu/tests/vsock_seqpacket_test.py (hunk 4 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py]: Updated. Thanks.

File: qemu/tests/vsock_seqpacket_test.py (hunk 3 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py:4]: Updated.

File: qemu/tests/cfg/vsock.cfg (hunk 2 above)
Comment (qcheng-redhat) [reply qemu/tests/cfg/vsock.cfg]: Add --depth=1. Thanks.

File: qemu/tests/vsock_seqpacket_test.py (hunk 1 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py]: Updated.

File: qemu/tests/cfg/vsock.cfg [hunk 6]
```diff
@@ -43,3 +43,15 @@
             ppc64, ppc64le:
                 virtio_dev_disable_legacy = on
                 virtio_dev_disable_modern = off
+        - seqpacket_test:
+             no RHEL.6 RHEL.7 RHEL.8
+             type = vsock_seqpacket_test
+             only q35
```
Comment (zhenyzha) [inline qemu/tests/cfg/vsock.cfg:49]: need add aarch64

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.seqpacket_test.arm64-pci: PASS (454.26 s)

**Comment by zhenyzha** [comment]
(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.seqpacket_test.arm64-pci: PASS (153.68 s)

LGTM.Acked-by: Zhenyu Zhang <[zhenyzha@redhat.com](mailto:zhenyzha@redhat.com)>

**Review by zhenyzha — APPROVED** [review]

**Comment by qcheng-redhat** [comment]
Hi @PaulYuuu @zhenyzha, Patch updated, please help review when you have time. 

 (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.seqpacket_test.q35: STARTED
| (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.seqpacket_test.q35: PASS (50.36 s)

**Comment by zhenyzha** [comment]
Tested with latest code pass

`(1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.vsock_test.seqpacket_test.arm64-pci: PASS (144.18 s)`

File: qemu/tests/vsock_seqpacket_test.py [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+def copy_compile_testsuite(test, vm, session):
+    """
+    Download and compile upstream test suite on guest and host.
+
+    :param test: QEMU test object
+    ::param vm: Object qemu_vm.VM
+    :param session: Guest session object
+    """
+    vsock_test_base_dir = "/home/"
```
Comment (PaulYuuu) [inline qemu/tests/vsock_seqpacket_test.py:23]: Maybe can save it into the `test.workdir` or `test.tmpdir`?

File: qemu/tests/vsock_seqpacket_test.py [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    host_ip = utils_net.get_host_ip_address(params)
+    guest_ip = vm.get_address()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    vsock_dev = params.objects("vsocks")[0]
```
Comment (PaulYuuu) [inline qemu/tests/vsock_seqpacket_test.py:71]: Only one vsock device in the cfg?
```suggestion
    vsock_dev = params["vsocks"]
```

File: qemu/tests/vsock_seqpacket_test.py (hunk 7 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py:23]: Considering use same directory for both host and guest for the following operations, keep use /home here.

File: qemu/tests/vsock_seqpacket_test.py (hunk 8 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py]: Updated

File: qemu/tests/cfg/vsock.cfg [hunk 9]
```diff
@@ -43,3 +43,15 @@
             ppc64, ppc64le:
                 virtio_dev_disable_legacy = on
                 virtio_dev_disable_modern = off
+        - seqpacket_test:
+             no RHEL.6 RHEL.7 RHEL.8
+             type = vsock_seqpacket_test
+             only q35, aarch64
+             host_cid = 2
+             disable_firewall = "service iptables stop; systemctl stop firewalld.service"
```
[the remark this answers is not in the document]
Comment (qcheng-redhat) [reply qemu/tests/cfg/vsock.cfg]: Updated.

File: qemu/tests/vsock_seqpacket_test.py [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+    host_ip = utils_net.get_host_ip_address(params)
+    guest_ip = vm.get_address()
+
+    disable_firewall = params.get("disable_firewall")
+    session.cmd(disable_firewall, ignore_all_errors=True)
+
+    vsock_dev = params["vsocks"]
+    guest_cid = vm.devices.get(vsock_dev).get_param("guest-cid")
+    host_cid = params["host_cid"]
+    port = random.randrange(1, 6000)
```
Comment (PaulYuuu) [inline qemu/tests/vsock_seqpacket_test.py:74]: If this is to find a free port on the host, I would suggest use `utils_misc.find_free_ports()`

**Review by PaulYuuu — COMMENTED** [review]
Others LGTM.

File: qemu/tests/vsock_seqpacket_test.py (hunk 10 above)
Comment (qcheng-redhat) [reply qemu/tests/vsock_seqpacket_test.py]: Updated.  @PaulYuuu 

 (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.seqpacket_test.q35: STARTED
- (1/1) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.x86_64.io-github-autotest-qemu.vsock_test.seqpacket_test.q35: PASS (47.61 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**